src/m_menu.c

```diff
@@ -3543,14 +3546,20 @@ setup_menu_t comp_settings1[] =  // Compatibility Settings screen #1
   {"Direct Vertical Aiming", S_YESNO|S_STRICT|S_CRITICAL, m_null, M_X, M_SPC,
    {"direct_vertical_aiming"}},
 
+  {"Auto Strafe 50", S_YESNO|S_STRICT|S_CRITICAL, m_null, M_X, M_SPC,
+   {"autostrafe50"}},
```

Comment:
Make sure to call `G_UpdateSideMove()` to actually update the function pointers.

```diff
diff --git a/src/m_menu.c b/src/m_menu.c
index cac0caf..5d45016 100644
--- a/src/m_menu.c
+++ b/src/m_menu.c
@@ -3550,7 +3550,7 @@ setup_menu_t comp_settings1[] =  // Compatibility Settings screen #1
    {"direct_vertical_aiming"}},
 
   {"Auto Strafe 50", S_YESNO|S_STRICT|S_CRITICAL, m_null, M_X, M_SPC,
-   {"autostrafe50"}},
+   {"autostrafe50"}, 0, G_UpdateSideMove},
 
   {"Pistol Start", S_YESNO|S_STRICT|S_CRITICAL, m_null, M_X, M_SPC,
    {"pistolstart"}},
```

Reply:
Here too:

```diff
diff --git a/src/m_menu.c b/src/m_menu.c
index cac0caf..c794d0b 100644
--- a/src/m_menu.c
+++ b/src/m_menu.c
@@ -3533,6 +3533,12 @@ static void M_UpdateCriticalItems(void)
                comp_settings1[comp1_blockmapfix]);
 }
 
+static void M_UpdateStrictMode(void)
+{
+  G_UpdateSideMove();
+  // anything else that isn't updated when a new map loads
+}
+
 setup_menu_t comp_settings1[] =  // Compatibility Settings screen #1
 {
   {"Compatibility", S_SKIP|S_TITLE, m_null, M_X, M_Y},
@@ -3540,7 +3546,8 @@ setup_menu_t comp_settings1[] =  // Compatibility Settings screen #1
   {"Default Compatibility Level", S_CHOICE|S_LEVWARN, m_null, M_X, M_SPC,
    {"default_complevel"}, 0, NULL, str_default_complevel},
 
-  {"Strict Mode", S_YESNO|S_LEVWARN, m_null, M_X, M_SPC, {"strictmode"}},
+  {"Strict Mode", S_YESNO|S_LEVWARN, m_null, M_X, M_SPC,
+   {"strictmode"}, 0, M_UpdateStrictMode},
 
   {"", S_SKIP, m_null, M_X, M_SPC},
 ```

Alternatively:

```diff
diff --git a/src/g_game.c b/src/g_game.c
index 4091715..adc9d11 100644
--- a/src/g_game.c
+++ b/src/g_game.c
@@ -3285,6 +3285,7 @@ void G_ReloadDefaults(boolean keep_demover)
   if (M_CheckParm("-strict"))
     strictmode = true;
 
+  G_UpdateSideMove();
   P_UpdateDirectVerticalAiming();
 
   pistolstart = default_pistolstart;
```

Reply:
Fixed here: https://github.com/fabiangreffrath/woof/pull/1402/commits/8702f98d195908efb457725d6e1448068cbc94b0